src/app/common/stores/applications/SsvWeb/Operator.store.ts

```diff
@@ -285,44 +287,47 @@ class OperatorStore extends BaseStore {
    * update operator address whitelist
    */
   async updateOperatorAddressWhitelist(operatorId: string, address: string) {
-    const applicationStore: ApplicationStore = this.getStore('Application');
-    const notificationsStore: NotificationsStore = this.getStore('Notifications');
     return new Promise(async (resolve) => {
       try {
         const walletStore: WalletStore = this.getStore('Wallet');
+        const applicationStore: ApplicationStore = this.getStore('Application');
+        const gasLimit = getFixedGasLimit(GasGroup.DECLARE_OPERATOR_FEE);
         const contractInstance = walletStore.setterContract;
-        const tx = await contractInstance.setOperatorWhitelist(operatorId, address);
-        if (tx.hash) {
-          applicationStore.txHash = tx.hash;
-          applicationStore.showTransactionPendingPopUp(true);
-        }
-        const receipt = await tx.wait();
-        if (receipt.blockHash) {
-          const event: boolean = receipt.hasOwnProperty('events');
-          if (event) {
-            let iterations = 0;
-            while (iterations <= MyAccountStore.CHECK_UPDATES_MAX_ITERATIONS) {
-              if (iterations >= MyAccountStore.CHECK_UPDATES_MAX_ITERATIONS) {
-                await this.refreshOperatorsAndClusters(resolve, true);
-                break;
-              }
-              iterations += 1;
-              const operator = await Operator.getInstance().getOperator(operatorId);
-              const changed = operator.address_whitelist.toString() === address.toString();
-              if (changed) {
-                await this.refreshOperatorsAndClusters(resolve, true);
-                break;
-              } else {
-                console.log('Operator is still not updated in API..');
+        await contractInstance.methods.setOperatorWhitelist(operatorId, address).send({ from: walletStore.accountAddress, gas: gasLimit })
+          .on('receipt', async (receipt: any) => {
+            const event: boolean = receipt.hasOwnProperty('events');
+            if (event) {
+              let iterations = 0;
+              while (iterations <= MyAccountStore.CHECK_UPDATES_MAX_ITERATIONS) {
```

Comment:
this while loop and all while loops structured the same are meaningless
code after if statement is unreachable - thus redundant. Please find every appearance of this code structure and simplify it to a single call to `await this.refreshOperatorsAndClusters(resolve, true);`
for the reference see this link [break in while loop](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/break#examples)